File: forkjoin-apikit-example/src/test/js/model/TestObject.js
'use strict';



import User from './User'


/**
 * @author  zuoge85 on 15/6/17.
*/
class TestObject {

    testForm:TestForm;

    id:String;

	/**
	 * 签名
	 * @see java.lang
	 * java.lang
	 */
    booleanValue:Boolean;

    intValue:Number;

    longValue:Number;

    floatValue:Number;

    doubleValue:Number;

    stringValue:String;

    bytesValue:Number[];

    regDate:Date;

    booleanValueArray:Boolean[];

    intValueArray:Number[];

    longValueArray:Number[];

    floatValueArray:Number[];

    doubleValueArray:Number[];

    stringValueArray:String[];

    regDateArray:Date[];

    user:User;

    users:User[];

    generics:Object[];

    genericObjs:TestObject[];

    genericUsers:TestObject[];

    genericObj:TestObject;

    generic:Object;
    constructor() {

    }

    formObject({testForm,id,booleanValue,intValue,longValue,floatValue,doubleValue,stringValue,bytesValue,regDate,booleanValueArray,intValueArray,longValueArray,floatValueArray,doubleValueArray,stringValueArray,regDateArray,user,users,generics,genericObjs,genericUsers,genericObj,generic}):TestObject{
        this.testForm = testForm;
        this.id = id;
        this.booleanValue = booleanValue;
        this.intValue = intValue;
        this.longValue = longValue;
        this.floatValue = floatValue;
        this.doubleValue = doubleValue;
        this.stringValue = stringValue;
        this.bytesValue = bytesValue;
        this.regDate = regDate;
        this.booleanValueArray = booleanValueArray;
        this.intValueArray = intValueArray;
        this.longValueArray = longValueArray;
        this.floatValueArray = floatValueArray;
        this.doubleValueArray = doubleValueArray;
        this.stringValueArray = stringValueArray;
        this.regDateArray = regDateArray;
        this.user = user;
        this.users = users;
        this.generics = generics;
        this.genericObjs = genericObjs;
        this.genericUsers = genericUsers;
        this.genericObj = genericObj;
        this.generic = generic;
        return this;
    }

    static of({testForm,id,booleanValue,intValue,longValue,floatValue,doubleValue,stringValue,bytesValue,regDate,booleanValueArray,intValueArray,longValueArray,floatValueArray,doubleValueArray,stringValueArray,regDateArray,user,users,generics,genericObjs,genericUsers,genericObj,generic}):TestObject{
        return new TestObject().formObject({testForm,id,booleanValue,intValue,longValue,floatValue,doubleValue,stringValue,bytesValue,regDate,booleanValueArray,intValueArray,longValueArray,floatValueArray,doubleValueArray,stringValueArray,regDateArray,user,users,generics,genericObjs,genericUsers,genericObj,generic});
    }

    static form(testForm:TestForm,id:String,booleanValue:Boolean,intValue:Number,longValue:Number,floatValue:Number,doubleValue:Number,stringValue:String,bytesValue:Number[],regDate:Date,booleanValueArray:Boolean[],intValueArray:Number[],longValueArray:Number[],floatValueArray:Number[],doubleValueArray:Number[],stringValueArray:String[],regDateArray:Date[],user:User,users:User[],generics:Object[],genericObjs:TestObject[],genericUsers:TestObject[],genericObj:TestObject,generic:Object):TestObject{
        return new TestObject().formObject({testForm,id,booleanValue,intValue,longValue,floatValue,doubleValue,stringValue,bytesValue,regDate,booleanValueArray,intValueArray,longValueArray,floatValueArray,doubleValueArray,stringValueArray,regDateArray,user,users,generics,genericObjs,genericUsers,genericObj,generic});
    }
}

export default TestObject;

File: forkjoin-apikit-example/src/main/java/org/forkjoin/apikit/example/model/TestObject.java
package org.forkjoin.apikit.example.model;


import org.forkjoin.apikit.core.Message;
import org.forkjoin.apikit.example.form.TestForm;

import java.util.Date;

/**
 * @author zuoge85 on 15/6/17.
 */
@Message
public class TestObject<T> {
    private String id;
    /**
     * 签名
     *
     * @see java.lang
     * @see java.lang
     */
    private boolean booleanValue;
    private int intValue;
    private long longValue;
    private float floatValue;
    private double doubleValue;


    private String stringValue;
    private byte[] bytesValue;

    private  Date regDate;


    private boolean[] booleanValueArray;
    private int[] intValueArray;
    private long[] longValueArray;
    private float[] floatValueArray;
    private double[] doubleValueArray;


    private String[] stringValueArray;

    private Date[] regDateArray;

    private User user;
    private User[] users;

    private T[] generics;

    private TestObject<T>[] genericObjs;

    private TestObject<User>[] genericUsers;

    private TestObject<T> genericObj;

    private T generic;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public boolean isBooleanValue() {
        return booleanValue;
    }

    public void setBooleanValue(boolean booleanValue) {
        this.booleanValue = booleanValue;
    }

    public int getIntValue() {
        return intValue;
    }

    public void setIntValue(int intValue) {
        this.intValue = intValue;
    }

    public long getLongValue() {
        return longValue;
    }

    public void setLongValue(long longValue) {
        this.longValue = longValue;
    }

    public float getFloatValue() {
        return floatValue;
    }

    public void setFloatValue(float floatValue) {
        this.floatValue = floatValue;
    }

    public double getDoubleValue() {
        return doubleValue;
    }

    public void setDoubleValue(double doubleValue) {
        this.doubleValue = doubleValue;
    }

    public String getStringValue() {
        return stringValue;
    }

    public void setStringValue(String stringValue) {
        this.stringValue = stringValue;
    }

    public byte[] getBytesValue() {
        return bytesValue;
    }

    public void setBytesValue(byte[] bytesValue) {
        this.bytesValue = bytesValue;
    }

    public Date getRegDate() {
        return regDate;
    }

    public void setRegDate(Date regDate) {
        this.regDate = regDate;
    }

    public boolean[] getBooleanValueArray() {
        return booleanValueArray;
    }

    public void setBooleanValueArray(boolean[] booleanValueArray) {
        this.booleanValueArray = booleanValueArray;
    }

    public int[] getIntValueArray() {
        return intValueArray;
    }

    public void setIntValueArray(int[] intValueArray) {
        this.intValueArray = intValueArray;
    }

    public long[] getLongValueArray() {
        return longValueArray;
    }

    public void setLongValueArray(long[] longValueArray) {
        this.longValueArray = longValueArray;
    }

    public float[] getFloatValueArray() {
        return floatValueArray;
    }

    public void setFloatValueArray(float[] floatValueArray) {
        this.floatValueArray = floatValueArray;
    }

    public double[] getDoubleValueArray() {
        return doubleValueArray;
    }

    public void setDoubleValueArray(double[] doubleValueArray) {
        this.doubleValueArray = doubleValueArray;
    }

    public String[] getStringValueArray() {
        return stringValueArray;
    }

    public void setStringValueArray(String[] stringValueArray) {
        this.stringValueArray = stringValueArray;
    }

    public Date[] getRegDateArray() {
        return regDateArray;
    }

    public void setRegDateArray(Date[] regDateArray) {
        this.regDateArray = regDateArray;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public User[] getUsers() {
        return users;
    }

    public void setUsers(User[] users) {
        this.users = users;
    }

    public T[] getGenerics() {
        return generics;
    }

    public void setGenerics(T[] generics) {
        this.generics = generics;
    }

    public TestObject<T>[] getGenericObjs() {
        return genericObjs;
    }

    public void setGenericObjs(TestObject<T>[] genericObjs) {
        this.genericObjs = genericObjs;
    }

    public TestObject<User>[] getGenericUsers() {
        return genericUsers;
    }

    public void setGenericUsers(TestObject<User>[] genericUsers) {
        this.genericUsers = genericUsers;
    }

    public TestObject<T> getGenericObj() {
        return genericObj;
    }

    public void setGenericObj(TestObject<T> genericObj) {
        this.genericObj = genericObj;
    }

    public T getGeneric() {
        return generic;
    }

    public void setGeneric(T generic) {
        this.generic = generic;
    }
}
